### PR TITLE
add a check for isAlive

### PR DESCRIPTION
**Pull request by U5B**
I believe this will prevent the bot from yeeting itself into the void on death (if the server doesn't lag him back)

### CODE REVIEW

**Comment by rom1504** [comment]
does it work? what's the difference before/after ?

File: lib/plugins/physics.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const { Vec3 } = require('vec3')
+const Vec3 = require('vec3').Vec3
```
Comment (Karang) [inline lib/plugins/physics.js:1]: Change not needed

File: lib/plugins/physics.js [hunk 2]
```diff
@@ -142,7 +142,7 @@ function inject (bot, { physicsEnabled }) {
       sendPacketPosition(position, onGround)
     } else if (lookUpdated && bot.isAlive) {
       sendPacketLook(yaw, pitch, onGround)
-    } else if (performance.now() - lastSent.time >= 1000) {
+    } else if (performance.now() - lastSent.time >= 1000 && bot.isAlive) {
```
Comment (Karang) [inline lib/plugins/physics.js:145]: Did you check with a proxy that it is vanilla behavior ? I'm pretty sure this packet need to be sent no matter if the bot is alive.

File: lib/plugins/physics.js [hunk 3]
```diff
@@ -245,6 +245,11 @@ function inject (bot, { physicsEnabled }) {
 
   // player position and look (clientbound)
   bot._client.on('position', (packet) => {
+    if (doPhysicsTimer === null) {
+      lastPhysicsFrameTime = performance.now()
+      doPhysicsTimer = setInterval(doPhysics, PHYSICS_INTERVAL_MS)
+    }
+
```
Comment (Karang) [inline lib/plugins/physics.js:252]: why move this code ?

Comment (U5B) [reply lib/plugins/physics.js:252]: Did not mean to move this.

File: lib/plugins/physics.js (hunk 2 above)
Comment (U5B) [reply lib/plugins/physics.js:145]: I believe this is vanilla behavior. For the duration that the bot is dead, it does not send a position packet.

**Comment by U5B** [comment]
> does it work? what's the difference before/after ?
https://ptb.discord.com/channels/413438066984747026/519952494768685086/891867964251447306

In laggy condtions (where the server is lagging) the bot will sometimes yeet itself into the void if the server does not correct the position.

**Comment by U5B** [comment]
The other thing is that we could do is disable physics simulation until we are "alive", (by adding '&& bot.isAlive') to https://github.com/U5B/mineflayer/blob/8752e5fe9680efbdcb67daf62c1f2502e24ef71f/lib/plugins/physics.js#L62.

**Comment by U5B** [comment]
Actually this probably isn't the correct fix, I will research this more later.

**Pull request closed without merging**